### ISSUE TITLE
table.decimal() shouldn't add precision and scale defaults

### ISSUE THREAD

**Issue by riteable** [knex/knex#2231]
PostgreSQL allows DECIMAL/NUMERIC types to be declared without precision and scale parameters. When scale is specified, the number is zero-padded at the right side. This is undesired in most cases.

Right now I have a column declared as `decimal(36, 18)`. This causes numbers such as 1.25 to be inserted as 1.250000000000000000. The values need to be formatted every time they're fetched from the DB which is pretty annoying.

So, as the title says: Knex should allow for `decimal()` to be declared without precision and scale parameters.

**Comment by elhigu** [comment]
I don't think that implementing this is worth of breaking backwards compatibility (docs say that currently default precision is 8 and scale is 2. I suppose that one can declare decimal without precision currently with `.specificType`.

If some one come up backwards compatible way of doing this and likes to implement it I'm ok with that though.

**Comment by riteable** [comment]
An option might be to be able to pass `false` for those params, because at the moment the arguments are being checked for `null`, `undefined`, and `NaN`.

**Comment by elhigu** [comment]
Maybe an options object `{ noPrecision: true }`. That way API won't seem so hacky.

**Comment by A-Tokyo** [comment]
@riteable @elhigu 

You can use `decimal(null);`
Specifying NULL as precision creates a decimal column that can store numbers of any precision and scale. (Only supported for Oracle, SQLite, Postgres)

**Comment by mercmobily** [comment]
Thanks for raising this issue about creating decimal columns without precision or scale.
You can already create decimal columns without precision/scale by passing `null`:

```javascript
table.decimal('amount', null);
```

- Issue #2231 (this issue) was created by elhigu on September 20, 2017
- The null functionality was added via PR #2353 on November 30, 2017 (about 2 months later)
- The commit fbf371f4e by Ivo Meißner implemented this feature

This is documented and works for PostgreSQL, Oracle, and SQLite. It creates a column that accepts any precision and scale.

MySQL and MSSQL don't support unlimited precision decimals, so for those use:

```javascript
table.specificType('amount', 'DECIMAL(65,30)'); // or whatever precision you need
```

I don't think we should change the API, as the current solution is sufficient:

- `table.decimal('amount')` → defaults to (8,2) for backwards compatibility
- `table.decimal('amount', null)` → unlimited precision where supported
- `table.specificType('amount', 'DECIMAL')` → full control

Adding an options object like `{ noPrecision: true }` would just be another way to do the same thing without meaningful benefit.
